From 66f34dd3ef1d0ea26213cf44dad97dbaabcd15ef Mon Sep 17 00:00:00 2001
From: Jacob Oursland <jacob.oursland@gmail.com>
Date: Fri, 18 Apr 2025 01:07:09 -0700
Subject: [PATCH] Add MSYS as a Windows platform.

---
 python/make_sdist.sh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/python/make_sdist.sh b/python/make_sdist.sh
index da4c3260..0eec1a9b 100755
--- a/python/make_sdist.sh
+++ b/python/make_sdist.sh
@@ -21,7 +21,7 @@ fi
 # Figure out the path to this script (https://stackoverflow.com/a/246128).
 package_dir="$(cd -- "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
 
-if [[ "$(uname)" == CYGWIN* || "$(uname)" == MINGW* ]]; then
+if [[ "$(uname)" == CYGWIN* || "$(uname)" == MINGW* || "$(uname)" == MSYS* ]]; then
   package_dir="$(cygpath -m ${package_dir})"
   readonly tmp_dir="$(TMPDIR="${LOCALAPPDATA//\\/$'/'}/Temp" mktemp -d)"
 else
@@ -35,7 +35,7 @@ cp -r "${package_dir}"/* .
 
 # Generate header files.
 old_pythonpath="${PYTHONPATH}"
-if [[ "$(uname)" == CYGWIN* || "$(uname)" == MINGW* ]]; then
+if [[ "$(uname)" == CYGWIN* || "$(uname)" == MINGW* || "$(uname)" == MSYS* ]]; then
   export PYTHONPATH="${old_pythonpath};${package_dir}/mujoco/python/.."
 else
   export PYTHONPATH="${old_pythonpath}:${package_dir}/mujoco/python/.."
-- 
2.49.0

