From 4aafc33a5d361a300f4483eab4368c9d5c266585 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Mon, 21 Oct 2024 22:04:15 +0200
Subject: [PATCH] Allow setting MUJOCO_GL to osmesa for Windows and macOS

---
 python/mujoco/gl_context.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/python/mujoco/gl_context.py b/python/mujoco/gl_context.py
index 61b861ecea..ff964dde16 100644
--- a/python/mujoco/gl_context.py
+++ b/python/mujoco/gl_context.py
@@ -23,9 +23,9 @@
 _SYSTEM = platform.system()
 _MUJOCO_GL = os.environ.get('MUJOCO_GL', '').lower().strip()
 if _MUJOCO_GL not in ('disable', 'disabled', 'off', 'false', '0'):
-  _VALID_MUJOCO_GL = ('enable', 'enabled', 'on', 'true', '1' , 'glfw', '')
+  _VALID_MUJOCO_GL = ('enable', 'enabled', 'on', 'true', '1' , 'glfw', '', 'osmesa')
   if _SYSTEM == 'Linux':
-    _VALID_MUJOCO_GL += ('glx', 'egl', 'osmesa')
+    _VALID_MUJOCO_GL += ('glx', 'egl')
   elif _SYSTEM == 'Windows':
     _VALID_MUJOCO_GL += ('wgl',)
   elif _SYSTEM == 'Darwin':
@@ -34,7 +34,7 @@
     raise RuntimeError(
         f'invalid value for environment variable MUJOCO_GL: {_MUJOCO_GL}')
 
-  if _SYSTEM == 'Linux' and _MUJOCO_GL == 'osmesa':
+  if _MUJOCO_GL == 'osmesa':
     from mujoco.osmesa import GLContext as _GLContext
     GLContext = _GLContext
   elif _SYSTEM == 'Linux' and _MUJOCO_GL == 'egl':
