From b0baecd8831a52af6c160cc444d2e3fba6136043 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Thu, 23 Jun 2022 17:55:30 +0200
Subject: [PATCH] Hardcode Python bindings compilation threads

---
 python/setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/setup.py b/python/setup.py
index d6291fd..0e777e3 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -240,7 +240,7 @@ def _configure_cmake(self):
 
     print('Building all extensions with CMake')
     subprocess.check_call(
-        [cmake, '--build', '.', f'-j{os.cpu_count()}', '--config', build_cfg],
+        [cmake, '--build', '.', f'-j2', '--config', build_cfg],
         cwd=self.build_temp)
 
   def build_extension(self, ext):
