diff --git a/python/setup.py b/python/setup.py
index 610b85816..120529c67 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -268,7 +268,7 @@ def _configure_cmake(self):
 
     print('Building all extensions with CMake')
     subprocess.check_call(
-        [cmake, '--build', '.', f'-j{os.cpu_count()}', '--config', build_cfg],
+        [cmake, '--build', '.', '-j1', '--config', build_cfg],
         cwd=self.build_temp)
 
   def build_extension(self, ext):
