From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio.traversaro@iit.it>
Date: Sun, 8 Jun 2025 09:21:38 +0200
Subject: [PATCH] workaround_cmake_19072

---
 simulate/CMakeLists.txt | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/simulate/CMakeLists.txt b/simulate/CMakeLists.txt
index ff64d734..c198e6cb 100644
--- a/simulate/CMakeLists.txt
+++ b/simulate/CMakeLists.txt
@@ -114,8 +114,12 @@ if(APPLE)
 endif()
 target_include_directories(
   platform_ui_adapter PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
-                             $<TARGET_PROPERTY:glfw,INTERFACE_INCLUDE_DIRECTORIES>
 )
+# Upstream MuJoCo is carefully not linking glfw here, and so it is just include its directories
+# to avoid dealing with two copies of glfw. In conda-forge (or in any C++/Python distribution)
+# pyglfw will depend on the same glfw copy linked here, so we can just link glfw normally
+# This avoids the tricky cmake bug https://gitlab.kitware.com/cmake/cmake/-/issues/19072
+target_link_libraries(platform_ui_adapter PUBLIC glfw)
 target_link_libraries(platform_ui_adapter PUBLIC mujoco::mujoco)
 if(SIMULATE_GLFW_DYNAMIC_SYMBOLS)
   target_compile_definitions(platform_ui_adapter PUBLIC mjGLFW_DYNAMIC_SYMBOLS)
-- 
2.43.0

