From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio.traversaro@iit.it>
Date: Sun, 8 Jun 2025 09:21:38 +0200
Subject: [PATCH] 619bis

---
 simulate/cmake/SimulateDependencies.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/simulate/cmake/SimulateDependencies.cmake b/simulate/cmake/SimulateDependencies.cmake
index dbbcd244..97f9303b 100644
--- a/simulate/cmake/SimulateDependencies.cmake
+++ b/simulate/cmake/SimulateDependencies.cmake
@@ -98,7 +98,7 @@ if(MUJOCO_EXTRAS_STATIC_GLFW)
   unset(BUILD_SHARED_LIBS_OLD)
 endif()
 
-if(NOT SIMULATE_STANDALONE)
+if(NOT SIMULATE_STANDALONE AND NOT MUJOCO_SIMULATE_USE_SYSTEM_GLFW)
   target_compile_options(glfw PRIVATE ${MUJOCO_MACOS_COMPILE_OPTIONS})
   target_link_options(glfw PRIVATE ${MUJOCO_MACOS_LINK_OPTIONS})
 endif()
-- 
2.43.0

