diff --git a/simulate/CMakeLists.txt b/simulate/CMakeLists.txt
index 4a615018..104a76f3 100644
--- a/simulate/CMakeLists.txt
+++ b/simulate/CMakeLists.txt
@@ -80,6 +80,7 @@ if(MUJOCO_HARDEN)
   endif()
 endif()
 
+find_package(lodepng REQUIRED)
 # Fetch lodepng dependency.
 if(NOT TARGET lodepng)
   FetchContent_Declare(
@@ -124,7 +125,9 @@ if(SIMULATE_GLFW_DYNAMIC_SYMBOLS)
 endif()
 add_library(mujoco::platform_ui_adapter ALIAS platform_ui_adapter)
 
-add_library(libmujoco_simulate STATIC $<TARGET_OBJECTS:platform_ui_adapter> $<TARGET_OBJECTS:lodepng>)
+add_library(libmujoco_simulate $<TARGET_OBJECTS:platform_ui_adapter>)
+target_link_libraries(libmujoco_simulate PRIVATE lodepng)
+set_target_properties(libmujoco_simulate PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
 set_target_properties(libmujoco_simulate PROPERTIES OUTPUT_NAME simulate)
 add_library(mujoco::libmujoco_simulate ALIAS libmujoco_simulate)
 set_target_properties(libmujoco_simulate PROPERTIES PUBLIC_HEADER "simulate.h;platform_ui_adapter.h;glfw_adapter.h;glfw_dispatch.h;glfw_corevideo.h;array_safety.h")
