diff --git a/sample/CMakeLists.txt b/sample/CMakeLists.txt
index 144557dd..46535a94 100644
--- a/sample/CMakeLists.txt
+++ b/sample/CMakeLists.txt
@@ -42,7 +42,7 @@ set(_INSTALL_SAMPLES ON)
 if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
   set(SAMPLE_STANDALONE ON)
   # If standalone, do not install the samples.
-  set(_INSTALL_SAMPLES OFF)
+  #set(_INSTALL_SAMPLES OFF)
 endif()
 
 list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
@@ -119,7 +119,7 @@ if(APPLE AND MUJOCO_BUILD_MACOS_FRAMEWORKS)
 endif()
 
 if(_INSTALL_SAMPLES)
-
+  if(FALSE)
   include(TargetAddRpath)
 
   # Add support to RPATH for the samples.
@@ -136,6 +136,7 @@ if(_INSTALL_SAMPLES)
     DEPENDS
     MUJOCO_ENABLE_RPATH
   )
+  endif()
 
   install(
     TARGETS basic
diff --git a/simulate/CMakeLists.txt b/simulate/CMakeLists.txt
index aaede15c..5255405c 100644
--- a/simulate/CMakeLists.txt
+++ b/simulate/CMakeLists.txt
@@ -50,7 +50,7 @@ set(_INSTALL_SIMULATE ON)
 if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
   set(SIMULATE_STANDALONE ON)
   # If standalone, do not install the samples.
-  set(_INSTALL_SIMULATE OFF)
+  #set(_INSTALL_SIMULATE OFF)
 endif()
 
 list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
@@ -232,7 +232,7 @@ if(SIMULATE_BUILD_EXECUTABLE)
   endif()
 
   if(_INSTALL_SIMULATE)
-
+    if(FALSE)
     include(TargetAddRpath)
 
     # Add support to RPATH for the samples.
@@ -246,11 +246,13 @@ if(SIMULATE_BUILD_EXECUTABLE)
       DEPENDS
       MUJOCO_ENABLE_RPATH
     )
+    endif()
 
     if (MUJOCO_WITH_USD)
       target_compile_definitions(simulate PUBLIC mjUSEUSD)
 
       # Add support to RPATH for the samples.
+      if(FALSE)
       target_add_rpath(
         TARGETS
         simulate
@@ -261,6 +263,7 @@ if(SIMULATE_BUILD_EXECUTABLE)
         DEPENDS
         MUJOCO_ENABLE_RPATH
       )
+      endif()
     endif()
 
     install(
