diff --git a/cmake/MujocoDependencies.cmake b/cmake/MujocoDependencies.cmake
index 9985af7..e285a33 100644
--- a/cmake/MujocoDependencies.cmake
+++ b/cmake/MujocoDependencies.cmake
@@ -158,7 +158,7 @@ if(NOT DEFINED CMAKE_POLICY_VERSION_MINIMUM)
 endif()
 findorfetch(
   USE_SYSTEM_PACKAGE
-  OFF
+  ON
   PACKAGE_NAME
   tinyobjloader
   LIBRARY_NAME
diff --git a/plugin/obj_decoder/CMakeLists.txt b/plugin/obj_decoder/CMakeLists.txt
index 689dfc5..5a909a3 100644
--- a/plugin/obj_decoder/CMakeLists.txt
+++ b/plugin/obj_decoder/CMakeLists.txt
@@ -12,7 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-target_compile_definitions(mujoco PRIVATE TINYOBJLOADER_IMPLEMENTATION)
 target_sources(mujoco PRIVATE
   obj_decoder.cc
 )
diff --git a/src/user/user_mesh.cc b/src/user/user_mesh.cc
index 9ebb688..7956375 100644
--- a/src/user/user_mesh.cc
+++ b/src/user/user_mesh.cc
@@ -36,9 +36,6 @@
 #include <mujoco/mjspec.h>
 #include "user/user_api.h"
 
-#ifdef MUJOCO_TINYOBJLOADER_IMPL
-  #define TINYOBJLOADER_IMPLEMENTATION
-#endif
 
 #if defined(__clang__)
   #pragma clang diagnostic push
